.github/workflows/e2e-connectivity.yaml

```
@@ -0,0 +1,222 @@
+        docker network connect "kind" $REGISTRY_NAME
+        # Make the $REGISTRY_NAME -> 127.0.0.1, to tell `ko` to publish to
+        # local reigstry, even when pushing $REGISTRY_NAME:$REGISTRY_PORT/some/image
+        sudo echo "127.0.0.1 $REGISTRY_NAME" | sudo tee -a /etc/hosts
+    
+    - name: Install Serving & Ingress
+      working-directory: ./src/knative.dev/serving
+      run: |
+        source ./test/e2e-common.sh
+        KIND=1
+        INGRESS_CLASS="${{ matrix.kingress }}.ingress.networking.knative.dev"
+        CLUSTER_DOMAIN="${{ matrix.cluster-suffix }}"
+        knative_setup
+        test_setup
+        echo "INGRESS_CLASS=$INGRESS_CLASS" >> $GITHUB_ENV
+        echo "CLUSTER_DOMAIN=$CLUSTER_DOMAIN" >> $GITHUB_ENV
+        echo "SYSTEM_NAMESPACE=$SYSTEM_NAMESPACE" >> $GITHUB_ENV
+        echo "GATEWAY_OVERRIDE=$GATEWAY_OVERRIDE" >> $GITHUB_ENV
+        echo "GATEWAY_NAMESPACE_OVERRIDE=$GATEWAY_NAMESPACE_OVERRIDE" >> $GITHUB_ENV
+    
+    - name: Apply NetworkPolicy
+      working-directory: ./src/knative.dev/serving
+      run: |
+        set -x
+        cat >> netpolicy.yaml <<EOF
+        apiVersion: networking.k8s.io/v1
+        kind: NetworkPolicy
+        metadata:
+          name: default-deny
+          namespace: $SYSTEM_NAMESPACE
+        spec:
+          podSelector:
+            matchExpressions:
+            - key: app
+              operator: In
+              values: [autoscaler, autoscaler-hpa]
+          policyTypes:
+          - Ingress
+          ingress:
+          - from:
+            - namespaceSelector:
+                matchLabels:
+                  # TODO: use this in preference to the projectcalico.org/name label after minVersion >= k8s 1.21.1
+                  kubernetes.io/metadata.name: $SYSTEM_NAMESPACE
+                  # When using tests against k8s versions lower than k8s 1.21.1 use the following label instead
+                  # projectcalico.org/name: $SYSTEM_NAMESPACE # calico specific
+        EOF
+        kubectl apply -f netpolicy.yaml
+    
+    - name: Check that autoscaler pods are unreachable
+      run: |
+        set -x
+        AUTOSCALER_IP=$(kubectl -n $SYSTEM_NAMESPACE get po -l 'app=autoscaler' -o jsonpath='{.items[*].status.podIPs[*].ip}')
+        # ping from inside the installation namespace should succeed
+        kubectl run pinger --rm -i --image=busybox --namespace=$SYSTEM_NAMESPACE --restart=Never -- ping -c 8 $AUTOSCALER_IP
+        PING_FAILED=false
+        function catch_ping_err {
+          if $PING_FAILED; then
+            echo "pinging autoscaler failed as expected"
+          else
+            echo "pinging autoscaler should have failed but succeeded"
+            exit 1
+          fi
+        }
+        trap catch_ping_err ERR
+        # ping from outside the installation namespace should fail
+        (kubectl run pinger --rm -i --image=busybox --restart=Never -- ping -c 8 $AUTOSCALER_IP || PING_FAILED=true)
```

I think this can be simpler

```suggestion
        # ping from outside the installation namespace should fail
        kubectl run pinger --rm -i --image=busybox --restart=Never -- ping -c 8 $AUTOSCALER_IP && echo "ping succeeded when it was supposed to fail" && exit 1
```